server/otc/ecs.go

```diff
@@ -239,11 +257,49 @@ func rebootECSHandler(c *gin.Context) {
 			return
 		}
 	}
-
 	c.JSON(http.StatusOK, common.ApiResponse{Message: "Reboot initiated."})
 	return
 }
 
+func validatePermissions(clients map[string]*gophercloud.ServiceClient, untrustedServers []servers.Server, username string) error {
+	groups, err := getGroups(username)
+	if err != nil {
+		return err
+	}
+	var allServers []servers.Server
+	for _, client := range clients {
+		serversInTenant, err := getServersByUsername(client, username, false)
+		if err != nil {
+			return err
+		}
+		allServers = append(allServers, serversInTenant...)
+	}
+	for _, untrustedServer := range untrustedServers {
+		// do not trust user data, because the metadata could have been modified
+		var server *servers.Server
+		for _, s := range allServers {
+			if untrustedServer.ID == s.ID {
+				if !cmp.Equal(untrustedServer.Metadata, s.Metadata) {
+					return fmt.Errorf("The server metadata couldn't be validated. Please try again.")
+				}
+				server = &s
+				break
+			}
+		}
+		if server == nil {
+			return fmt.Errorf("Server not found. Please try again.")
```

Comment:
Why should the user try again? If the request contains an invalid server it won't help to repeat the request, does it?
```suggestion
			return fmt.Errorf("Server not found.")
```

Reply:
updated